doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,565 @@
+error type. For more information on how it is used for error handling see
+[error-handling.md](../../design/error-handling.md).
+
+Assume we have ``E`` as an error type, then we can create a value
+```cpp
+iox::cxx::expected<int, E> result(iox::cxx::success<int>(73));
+```
+
+and use the value or handle a potential error
+```cpp
+if (!result.has_error())
+{
+    auto value = result.value();
+    // do something with the value
+}
+else
+{
+    auto error = result.get_error();
+    // handle the error
+}
+```
+
+Should we need an error value we set
+```cpp
+result = iox::cxx::error<E>(errorCode);
+```
+which assumes that E can be constructed from an ``errorCode``.
+
+We again can employ a functional approach like this
+```cpp
+auto handleValue = [](int& value) { /*do something with the value*/ };
+auto handleError = [](E& value) { /*handle the error*/ };
+result.and_then(handleValue).or_else(handleError);
+```
+
+There are more convenience functions such as ``value_or`` which provides the value or an alternative specified by the
+user. These can be found in [``expected.hpp``](../../../iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp).
+
+
+## Using the API
+
+Armed with the terminology and functional concepts, we can start to use the API to send and receive data. This
+involves settting up the runtime in each application, creating publishers in applications that need to send data and
+subscribers in applications that want to receive said data. An application can have both, publishers and subscribers.
+It can even send data to itself, but this usually makes little sense.
+
+### Initialize the runtime
+
+Create a runtime with a unique name among all applications for each application
+
+```cpp
+iox::runtime::PoshRuntime::initRuntime("some_unique_name");
+```
+
+Now this application is ready to communicate with the middleware daemon RouDi.
+
+### Defining a topic
+
+We need to define a data type we can send, which can be any struct or class or even a plain type, such as an int.
+
+```cpp
+struct CounterTopic
+{
+    CounterTopic(uint32_t counter = 0U)
+        : counter(counter)
+    {
+    }
+
+    uint32_t counter;
+};
+```
+
+The topic type must be default- and copy-constructible when the typed API is used. Using the untyped API imposes no
+such restriction, it just has to be possible to construct the data type at a given memory location.
+
+### Typed API
+
+We now demonstrate how to send and receive data with the typed API. 
+
+#### Creating a publisher
+
+We create a publisher that offers our CounterTopic.
+
+```cpp
+iox::popo::TypedPublisher<CounterTopic> publisher({"Group", "Instance", "CounterTopic"});
+publisher.offer();
+```
+
+Note that it suffices to set the first two identifiers (Group and Instance) to some default values for all topics.
+#### Sending data
+
+Now we can use the publisher to send the data in various ways.
+
+1. Loan and assign
+
+    ```cpp
+    auto result = publisher.loan();
```

Comment:
Something for 1.0.

There is some merit to leave the error checking to the user I think, because we can run out of samples if the mempools are not able to satisfy the requests due to usage by other publishers/subscribers. I.e. it might not be the fault of the terminating application which I consider kind of bad (in a "I did nothing wrong, why do I have to face harsh consequences?" way).

Reply:
I agree, added a todo in https://github.com/eclipse/iceoryx/issues/408#issue-758449131